packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -155,12 +157,15 @@ export class SerializedStateManager {
 		containerEvent: IEventProvider<ISerializerEvent>,
 		private readonly containerDirty: () => boolean,
 		private readonly supportGetSnapshotApi: () => boolean,
+		snapshotExpiryTimeout?: number,
 	) {
 		this.mc = createChildMonitoringContext({
 			logger: subLogger,
 			namespace: "serializedStateManager",
 		});
 
+		this.snapshotExpiryTimeout = snapshotExpiryTimeout ?? this.snapshotExpiryTimeout;
+		this.refreshTimer = new Timer(this.snapshotExpiryTimeout, () => this.tryRefreshSnapshot());
```

Comment:
the time should only run if offline is enabled

Reply:
Timer does not run until start or restart methods are called and those are inside offline enabled flows.